mysticeti-core/src/block_manager.rs

```diff
@@ -39,18 +39,32 @@ impl BlockManager {
         }
     }
 
+    /// Attempts to process (accept) the provided blocks and stores them only when the causal history
+    /// is already present. If a block can't be processed, then it is parked in the `blocks_pending` map
+    /// and any missing references are recorded in the `block_references_waiting` and in the `missing` vector.
+    /// The method returns a tuple where the newly accepted/processed blocks are returned and the missing references
+    /// of the provided blocks. Keep in mind that the missing references are returned only the first one for a specific
+    /// provided block. If we attempt to add the same block again, then its missing references won't be returned again.
```

Comment:
Is there a separate mechanism requesting the same missing blocks from a random peer? Basically I'm trying to understand if there is a liveness problem.

Reply:
Yes , there is nothing that will prevent the current `synchronizer` ask and fetch the missing blocks as dictated by the block manager - so there shouldn't be a liveness issue. Again, as the synchronizer is something that needs an overall refactoring I haven't paid attention now to duplication of work .